admin/UpdateDatabasePrivileges.pl

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env perl
+
+use warnings;
+use strict;
+
+use FindBin;
+use Getopt::Long qw( GetOptions );
+use lib "$FindBin::Bin/../lib";
+
+use MusicBrainz::Server::Constants qw( @FULL_SCHEMA_LIST );
+use MusicBrainz::Server::Context;
+use MusicBrainz::Server::Log qw( log_info );
+
+my $primary_ro_role = 'musicbrainz_ro';
+my @other_ro_roles = qw( caa_redirect sir );
+my $database = 'MAINTENANCE';
+my $grant_privileges = 1;
+
+my $help = <<EOF;
+Usage: UpdateDatabasePrivileges.pl [OPTIONS]
+
+By default, this command grants USAGE/SELECT privileges to a set of roles
+intended for read-only access. It revokes write privileges.
+
+If `--nogrant` is specified, then it also revokes read privileges
+(USAGE/SELECT).
+
+Options are:
+        --database          Database to connect to (default: MAINTENANCE)
+        --primary-ro-role   Name of the primary READONLY role
+                            (default: musicbrainz_ro)
+        --other-ro-role     Name of another role allowed RO access to the
+                            database (may be specified multiple times)
+                            (default: caa_redirect, sir)
+        --[no]grant         Whether to GRANT or REVOKE USAGE/SELECT privileges
+                            (default: GRANT)
+    -h, --help              Show this help
+
+EOF
+
+GetOptions(
+    'database=s'        => \$database,
+    'primary-ro-role=s' => \$primary_ro_role,
+    'other-ro-role=s'   => \@other_ro_roles,
+    'grant!'            => \$grant_privileges,
+    'help|h'            => sub { print $help; exit },
+) or exit 2;
+print($help), exit 2 if @ARGV;
+
+my $c = MusicBrainz::Server::Context->create_script_context(
+    database => 'SYSTEM_' . $database,
+);
+my $sql = $c->sql;
+
+$sql->auto_commit;
+my $existing_role_names = $sql->select_single_column_array(
+    <<~'SQL',
+    SELECT rolname
+      FROM pg_roles
+     WHERE rolname = any(?)
+    SQL
+    [$primary_ro_role, @other_ro_roles],
+);
+my %existing_role_names = map { $_ => 1 } @$existing_role_names;
+
+if ($existing_role_names{$primary_ro_role}) {
+    my $quoted_primary_role = $sql->dbh->quote_identifier($primary_ro_role);
+    $sql->begin;
+    for my $schema (@FULL_SCHEMA_LIST) {
```

Comment:
There should probably be a condition for each schema has it might not exist in the current database. (?)